### PR TITLE
parameterized test_graph_optims and test_graph_scaling_fused_optimizers

### LINKED ISSUE #123451: Update test_cuda.py and test_torch.py optim tests to use OptimizerInfo and optim_db
### 🚀 The feature, motivation and pitch

**Update: all tasks below have in progress PRs currently!**

In test_cuda.py and test_torch.py, we have the following optimizer tests that should be migrated to use the new OptimizerInfo infrastructure. (See any of the tests in TestOptimRenewed for how to apply OptimizerInfos.) The basic structure to expect for an updated test will look like:

```
    @optims([optim for optim in optim_db if ...], dtypes=[torch.float32])
    def test_name(self, device, dtype, optim_info):
        optim_cls = optim_info.optim_cls
        optim_inputs = optim_info.optim_inputs_func(device=device)
        for optim_input in optim_inputs:
            params = ...
            // processing, like updating grads

            optimizer = optim_cls(params, ..., **optim_input.kwargs)

            // the actual test
            ...
            self.assertEquals(...)
```

test_cuda.py:
- [ ] test_grad_scaling_autocast_fused_optimizers will be addressed in https://github.com/pytorch/pytorch/pull/124904 and by @SandishKumarHN 
- [x] test_graph_grad_scaling
- [ ] test_graph_optims
- [ ] test_graph_optims_with_explicitly_capturable_param_groups
- [ ] test_graph_scaling_fused_optimizers

test_torch.py: #125538
- [x] tests that call `_grad_scaling_autocast_test` can be combined into one test
- [x] test_params_invalidated_with_grads_invalidated_between_unscale_and_step should be tested on more configurations (like SGD fused!) 
- there are many other tests that test just one optimizer config that we can leave for now.

Feel free to take up one or a subset of those tests to migrate--edit your username next to the checklist to claim an item to submit PRs for!

### Alternatives

_No response_

### Additional context

_No response_

cc @vincentqb @jbschlosser @albanD @crcrpar

### PR DESCRIPTION
**Pull request by zero000064**
Fixes #123451

This is a rework of a reverted pull request, https://github.com/pytorch/pytorch/pull/125127.
The test failure is fixed.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @janeyx99

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
Thanks for this! Running CI now...

**Comment by zero000064** [comment]
@janeyx99 the workflow failed by timeout, could we rerun it?

**Review by janeyx99 — COMMENTED** [review]
Are there any differences between this and #125127?

File: test/test_cuda.py [hunk 1]
```diff
@@ -5069,10 +4868,180 @@ def create_mempool_and_make_active():
...
+        [
+            optim
+            for optim in optim_db
+            if "fused" in optim.supported_impls and "cuda" in optim.supports_fused_on
+        ],
+        dtypes=[torch.float32],
+    )
+    def test_graph_scaling_fused_optimizers(self, device, dtype, optim_info):
+        optim_cls = optim_info.optim_cls
+        print(f"optim_cls:{optim_cls}")
```
Comment (janeyx99) [inline test/test_cuda.py:4964]: remove
```suggestion
```

File: test/test_cuda.py [hunk 2]
```diff
@@ -5069,10 +4868,180 @@ def create_mempool_and_make_active():
...
+
+    @onlyCUDA
+    @unittest.skipIf(
+        not TEST_CUDA_GRAPH, "CUDA >= 11.0 or ROCM >= 5.3 required for graphs"
+    )
+    @optims(
+        [
+            optim
+            for optim in optim_db
+            if "fused" in optim.supported_impls and "cuda" in optim.supports_fused_on
```
Comment (janeyx99) [inline test/test_cuda.py:4958]: ah it's just this line

**Review by janeyx99 — APPROVED** [review]
Looks good, remove the print please!

**Comment by janeyx99** [comment]
We don't have to worry about the pr_time_benchmarks failure (looks like it's in the middle of getting set up). Once you remove, we can rebase and rerun CI to merge!

File: test/test_cuda.py (hunk 2 above)
Comment (zero000064) [reply test/test_cuda.py:4958]: Yes, only run tests for cuda-supported fused implementation.

**Comment by zero000064** [comment]
> Are there any differences between this and #125127?

Only run fused tests on cuda-supported fused impl.

File: test/test_cuda.py (hunk 1 above)
Comment (zero000064) [reply test/test_cuda.py]: removed.

**Comment by zero000064** [comment]
> We don't have to worry about the pr_time_benchmarks failure (looks like it's in the middle of getting set up). Once you remove, we can rebase and rerun CI to merge!

resolved all comments.

**Pull request closed without merging**